pkg/reconciler/configuration/configuration.go

```diff
@@ -335,6 +335,14 @@ func isRevisionStale(ctx context.Context, rev *v1alpha1.Revision, config *v1alph
 		if err.(v1alpha1.LastPinnedParseError).Type != v1alpha1.AnnotationParseErrorTypeMissing {
 			logger.Errorf("Failed to determine revision last pinned: %v", err)
 		}
+		// Revision was never pinned and its RevisionConditionReady is not true after staleRevisionCreateDelay.
+		// It usually happens when ksvc was deployed with wrong configuration.
+		if err.(v1alpha1.LastPinnedParseError).Type == v1alpha1.AnnotationParseErrorTypeMissing {
```

Comment:
Perhaps move this condition up and `logger.Errorf("Failed to determine revision last pinned: %v", err)` into its `else`?